R/estimate_vaccineff.R

```diff
@@ -122,20 +122,40 @@ summary.vaccineff <- function(object, ...) {
   stopifnot("Input must be an object of class 'vaccineff'" =
       checkmate::test_class(object, "vaccineff")
   )
-  cat(
-    sprintf("Vaccine Effectiveness at %i days computed as VE = 1 - HR:\n",
-            object$at)
+  summ <- list(
+    ve = object$ve,
+    at = object$at,
+    p_value_schoenfeld = object$cox_model$p_value
+  )
+  class(summ) <- "summary.vaccineff"
+  return(summ)
+}
+
+#' @title Print summary of VE Results
+#'
+#' @description Print summary of `vaccineff` object.
+#'
+#' @param x Object of the class `summary.vaccineff`.
+#' @param ... Additional arguments passed to other functions.
+#' @return Summary of the results from `estimate_vaccineff`.
+#' @export
+
+print.summary.vaccineff <- function(x, ...) {
+  message(
+    sprintf("Vaccine Effectiveness at %i days computed as VE = 1 - HR:",
+            x$at)
   )
 
-  print(object$ve)
-  cat("\nSchoenfeld test for Proportional Hazards assumption:\n")
-  cat(sprintf("p-value = %s\n", object$cox_model$p_value))
-  if (object$cox_model$p_value < 0.05) {
-    message <- paste0("\np-value < 0.05. Please check loglog plot",
+  print(x$ve, row.names = FALSE)
+  message("\nSchoenfeld test for Proportional Hazards assumption:")
+  message(sprintf("p-value = %s", x$p_value_schoenfeld))
+  if (x$p_value_schoenfeld < 0.05) {
+    warning_schoenfeld <- paste0("\np-value < 0.05. Please check loglog plot",
       " for Proportional Hazards assumption"
     )
-    warning(message)
+    message("\nWarning Message:", warning_schoenfeld)
```

Comment:
Is there a reason you're using `message()` over `warning()` here even though it is a warning message?

Reply:
I used `message()` because `warning()` creates an alert related to the print method (see below). However, the message returned is the log of alerts from the entire matching process.
```
Warning message:
In print.summary_vaccineff(x) : 
p-value < 0.05. Please check loglog plot for Proportional Hazards assumption
```